lua/entities/gmod_wire_expression2/init.lua

```diff
@@ -95,86 +97,92 @@ function ENT:Destruct()
 end
 
 function ENT:UpdatePerf()
-	if not self.context then return end
+	local context = self.context
+	if not context then return end
 	if self.error then return end
 
-	self.context.prfbench = self.context.prfbench * 0.95 + self.context.prf * 0.05
-	self.context.prfcount = self.context.prfcount + self.context.prf - e2_softquota
-	self.context.timebench = self.context.timebench * 0.95 + self.context.time * 0.05 -- Average it over the last 20 ticks
+	context.prfbench = context.prfbench * 0.95 + context.prf * 0.05
+	context.prfcount = context.prfcount + context.prf - e2_softquota
+	context.timebench = context.timebench * 0.95 + context.time * 0.05 -- Average it over the last 20 ticks
 
-	if self.context.prfcount < 0 then self.context.prfcount = 0 end
+	if context.prfcount < 0 then context.prfcount = 0 end
 
 	self:UpdateOverlay()
 
-	self.context.prf = 0
-	self.context.time = 0
-
+	context.prf = 0
+	context.time = 0
 end
 
 function ENT:Execute()
-	if self.error or not self.context or self.context.resetting then return end
+	local selfTbl = self:GetTable()
+	local context = selfTbl.context
+	if not context or selfTbl.error or context.resetting then return end
 
-	self:PCallHook('preexecute')
+	self:PCallHook("preexecute")
 
-	self.context.stackdepth = self.context.stackdepth + 1
+	local name = selfTbl.name
```

Comment:
This shouldn't be cached here since at worst it'll be used twice, and at best it'll never be used. It slightly impacts normal operation therefore.

Reply:
Should be dealt with now.